src/psyclone/tests/dependency_test.py

```diff
@@ -579,15 +579,22 @@ def test_lfric_stencils():
     assert "f2_stencil_dofmap: READ" in var_info
 
 
-def test_lfric_operator_different_spaces():
-    ''' Tests that implicit parameters for an operator with different to
-    and from spaces are created correctly.
+def test_lfric_various_basis():
+    ''' Tests that implicit parameters for various basis related
+    functionality work as expected.
 
     '''
     _, invoke_info = get_invoke("10.3_operator_different_spaces.f90",
                                 "dynamo0.3", idx=0)
     var_info = str(VariablesAccessInfo(invoke_info.schedule))
     assert "orientation_w2: READ" in var_info
+    assert "basis_w3_qr: READ" in var_info
+    assert "diff_basis_w0_qr: READ" in var_info
+    assert "diff_basis_w2_qr: READ" in var_info
+    assert "np_xy_qr: READ" in var_info
+    assert "np_z_qr: READ" in var_info
+    assert "weights_xy_qr: READ" in var_info
+    assert "weights_z_qr: READ" in var_info
```

Comment:
I think if you can extend this test to check whether the `VariablesAccessInfo` object has the correct scalar/array info for each of these then that should address some of my earlier comments.

Reply:
N/A anymore 😁